module/support/optimisation/WalkOptimiser/src/WalkOptimiser.cpp

```diff
@@ -129,36 +132,34 @@ namespace support {
                 }
             });
 
-            on < Trigger<FixedWalkFinished>,
-                <Sync<WalkOptimiser>>().then(
-                    "Walk Routine Finised",
-                    [this] {
-                        // Get and reset data
-                        fitnesses[currentSample] = data.popFitness();
-                        std::cerr << "Sample Done! Fitness: " << fitnesses[currentSample] << std::endl;
-                        if (currentSample >= samples.n_rows - 1) {
-                            emit(std::make_unique<OptimisationComplete>());
-                        }
-                        else {
-                            // Setup new parameters
-                            std::cerr << "Sample:" << ++currentSample << std::endl;
-                            setWalkParameters(getWalkConfig(samples.row(currentSample).t()));
-                            // Now wait for WalkConfigSaved
-                        }
-                    });
+            on<Trigger<FixedWalkFinished>, Sync<WalkOptimiser>>().then("Walk Routine Finised", [this] {
+                // Get and reset data
+                fitnesses[currentSample] = data.popFitness();
+                std::cerr << "Sample Done! Fitness: " << fitnesses[currentSample] << std::endl;
+                if (currentSample >= samples.rows() - 1) {
+                    emit(std::make_unique<OptimisationComplete>());
+                }
+                else {
+                    // Setup new parameters
+                    std::cerr << "Sample:" << ++currentSample << std::endl;
+                    setWalkParameters(getWalkConfig(samples.row(currentSample).transpose()));
+                    // Now wait for WalkConfigSaved
+                }
+            });
 
             on<Trigger<OptimisationComplete>, Sync<WalkOptimiser>>().then("Record Results", [this] {
                 // Combine samples
-                arma::vec result = utility::math::optimisation::PGA::updateEstimate(samples, fitnesses);
+                Eigen::VectorXd result;
+                // result = utility::math::optimisation::PGA::updateEstimate(samples, fitnesses);
```

Comment:
Pretty sure you've entirely broken this module.
So either this porting is not yet finished or you need to delete this module entirely

Reply:
No, you broke this when you purged all of optimisers